shell/browser/javascript_environment.cc

```diff
@@ -65,6 +66,83 @@ struct base::trace_event::TraceValue::Helper<
 
 namespace electron {
 
+class ArrayBufferAllocator : public v8::ArrayBuffer::Allocator {
```

Comment:
@nornagon what is the context of this change ?

Reply:
- [This CL](https://chromium-review.googlesource.com/c/chromium/src/+/2693619) enabled the partition allocator on Linux (it was already enabled on Windows, but not macOS. The test doesn’t fail on macOS.)
- The partition allocator [hard-crashes](https://source.chromium.org/chromium/chromium/src/+/master:base/allocator/partition_allocator/partition_bucket.cc;l=607-634;drc=062c315a858a87f834e16a144c2c8e9591af2beb) if you call calloc() with a size greater than the max (because it doesn’t pass the base::PartitionAllocReturnNull flag [here](https://source.chromium.org/chromium/chromium/src/+/master:base/allocator/allocator_shim_default_dispatch_to_partition_alloc.cc;l=259;drc=594527cfe905eb0f0d666a0bbcecac10d0110eed))
- The gin ArrayBuffer allocator [uses calloc](https://source.chromium.org/chromium/chromium/src/+/master:gin/array_buffer.cc;l=33;drc=947043888c9424f7032985dfc5a6b57702a44b57) to allocate zero-initialized backing stores for ArrayBuffers (there’s a TODO here from 2017(!) to do something different).